src/target/riscv/riscv-013.c

```diff
@@ -977,27 +977,51 @@ static int execute_abstract_command(struct target *target, uint32_t command,
 	return ERROR_OK;
 }
 
-static riscv_reg_t read_abstract_arg(struct target *target, unsigned index,
-		unsigned size_bits)
+static void add_data_regs_read(struct riscv_batch *batch, unsigned int index,
+		unsigned int size_bits)
 {
+	const unsigned int size_in_words = size_bits / 32;
+	assert(size_bits % 32 == 0);
+	const unsigned int offset = index * size_in_words;
+	for (unsigned int i = 0; i < size_in_words; ++i) {
+		const unsigned int reg_address = DM_DATA0 + offset + i;
+		riscv_batch_add_dm_read(batch, reg_address);
+	}
+}
+
+static riscv_reg_t read_data_regs(struct riscv_batch *batch,
+		unsigned int index, unsigned int size_bits)
+{
+	const unsigned int size_in_words = size_bits / 32;
+	assert(size_bits % 32 == 0);
+	assert(size_in_words * sizeof(uint32_t) <= sizeof(riscv_reg_t));
```

Comment:
I'd recommend to rewrite the assert to a more direct form, for example:

```suggestion
	assert(size_in_words == 1 || size_in_words == 2);
```
or

```suggestion
	assert(size_in_bits == 32 || size_in_bits == 64);
```

or, if you wish to check also the riscv_reg_t datatype (to prevent the unlikely of someone changing its size later):

```suggestion
	assert(size_in_words == 1 || size_in_words == 2);
	assert(sizeof(riscv_reg_t) == 8);	
```

Reply:
Is this really necessary?
I would like to avoid using arbitrary constants where possible (1 & 2, 32 & 64, 8).

Will address if you insist.

Reply:
The constants are not arbitrary - they are the list of allowed inputs to the function. The motivation for my suggestion is code readability.

> Is this really necessary?

No, but I believe it helps code readability :)

> Will address if you insist.

Please, choose the style yourself, I won't insist on this one.